### PR TITLE
Make expiration of currency networks optional

### PR DESCRIPTION
**Pull request by weilbith**
This includes:
- extend currency network contract to allow disabling freezing on initialization
- disable freezing per default for deploy tool

Closes: trustlines-network/project#682

### CODE REVIEW

File: py-deploy/tldeploy/cli.py [hunk 1]
```diff
@@ -99,9 +99,10 @@ def cli(ctx, version):
 @currency_network_contract_name_option
 @click.option(
     "--expiration-time",
-    help="Expiration time of the currency network after which it will be frozen",
+    help="Expiration time of the currency network after which it will be frozen (disabled by default)",
```
Comment (cducrest) [inline py-deploy/tldeploy/cli.py]: I would instead write `(0 means disabled)` since we show a default of 0 in the tool.

Comment (weilbith) [reply py-deploy/tldeploy/cli.py]: :heavy_check_mark:

File: py-deploy/tldeploy/cli.py [hunk 2]
```diff
@@ -143,14 +144,11 @@ def currencynetwork(
             "Prevent mediator interests is not necessary if custom interests are disabled."
         )
 
-    if expiration_date is not None and expiration_time is not None:
+    if expiration_date is not None and expiration_time != 0:
```
Comment (berndbohmeier) [inline py-deploy/tldeploy/cli.py]: Please have None as default. 
If you do it this way, someone could try to disable it with 0, but also accidentally provide a date

File: py-deploy/tldeploy/cli.py [hunk 3]
```diff
@@ -99,9 +99,10 @@ def cli(ctx, version):
 @currency_network_contract_name_option
 @click.option(
     "--expiration-time",
-    help="Expiration time of the currency network after which it will be frozen",
+    help="Expiration time of the currency network after which it will be frozen (0 means disabled)",
     required=False,
     type=int,
+    default=0,  # disabled
```
Comment (berndbohmeier) [inline py-deploy/tldeploy/cli.py]: Remove 0 as default

File: py-deploy/tldeploy/cli.py (hunk 2 above)
Comment (weilbith) [reply py-deploy/tldeploy/cli.py]: Okay make sense. :thinking: 
So to get then `0` as default for the deployment if the user does not define it, if would do something like:

```python
if expiration_time is None:
    expiration_time = 0
```

I'm thinking about the perspective of the user. `None` is the "default" which translates to `0` if the user defines nothing. This is the same as he would specifically set this parameter to `0`. Do we want to extend the help text for that information? Or is this just clear, no expiration date no freezing?

Comment (berndbohmeier) [reply py-deploy/tldeploy/cli.py]: Only set it to 0 if both are None. Just use the condition that was there before you removed it.

Comment (berndbohmeier) [reply py-deploy/tldeploy/cli.py]: Yes, I would add a help for both time and date, that explains the default is no freezing.

Comment (berndbohmeier) [reply py-deploy/tldeploy/cli.py]: > This is the same as he would specifically set this parameter to 0

No it is not the same, this is why you should not do it this way. 
0 means he wants it to be 0, so we should give him an error if he also provided a contradicting date arg.

Comment (weilbith) [reply py-deploy/tldeploy/cli.py]: Understood. I will do it this way. :+1:

File: py-deploy/tldeploy/cli.py (hunk 3 above)
Comment (weilbith) [reply py-deploy/tldeploy/cli.py]: :heavy_check_mark:

**Pull request merged**